docs: Add a "Troubleshooting span metrics high cardinality" section to `otelcol.connector.spanmetrics` docs (#5661)
grafana/alloy: docs/sources/reference/components/otelcol/otelcol.connector.spanmetrics.md

Beyond degrading APM interfaces with numerous non-meaningful operation names, this problem leads to metric time series explosion, resulting in significant performance degradation and increased costs.

`otelcol.connector.spanmetrics` provides an optional circuit breaker through the `aggregation_cardinality_limit` attribute (disabled by default) to mitigate cardinality explosion.
While this feature addresses performance and cost concerns, it does not resolve the underlying issue of semantically meaningless operation names.

### Fixing high cardinality span name issues

The ideal long-term solution is to modify the OpenTelemetry instrumentation code to comply with semantic conventions, preventing the generation of non-compliant high cardinality span names.

However, deploying updated instrumentation libraries can be time-consuming, often requiring an immediate interim solution to restore observability backend functionality.

#### Addressing high cardinality span names in the ingestion pipeline

An effective short-term solution is to implement a span sanitization layer within the observability ingestion pipeline.

> This can be achieved by using `otelcol.processor.transform`'s [`set_semconv_span_name()` function][set_semconv_span_name] immediately before `otelcol.connector.spanmetrics` to enforce semantic conventions on span names.

Reviewer: ```suggestion
An effective short-term solution is to implement a layer that sanitizes span names within the observability ingestion pipeline.
You can use `otelcol.processor.transform`'s [`set_semconv_span_name()` function][set_semconv_span_name] immediately before `otelcol.connector.spanmetrics` to enforce semantic conventions on span names.
```

These patterns render operation lists difficult to interpret and ineffective for monitoring purposes.

This issue stems from violations of [OpenTelemetry semantic conventions](https://opentelemetry.io/docs/specs/semconv/), which require span names to have low {{< term "cardinality" >}}cardinality{{< /term >}}.
Refer to [HTTP span name specs](https://opentelemetry.io/docs/specs/semconv/http/http-spans/#name) for more information.
Beyond degrading APM interfaces with numerous non-meaningful operation names, this problem leads to metric time series explosion, resulting in significant performance degradation and increased costs.

`otelcol.connector.spanmetrics` provides an optional circuit breaker through the `aggregation_cardinality_limit` attribute (disabled by default) to mitigate cardinality explosion.
While this feature addresses performance and cost concerns, it doesn't resolve the underlying issue of semantically meaningless operation names.

### Fix high cardinality span name issues

The ideal long-term solution is to modify the OpenTelemetry instrumentation code to comply with semantic conventions, preventing the generation of non-compliant high cardinality span names.

> However, deploying updated instrumentation libraries can be time-consuming, often requiring an immediate interim solution to restore observability backend functionality.

Reviewer: ```suggestion
However, deploying updated instrumentation libraries can be time-consuming.
You often need an immediate interim solution to restore observability backend functionality.
```

  //    Unsanitized span names, when different, are kept in the `unsanitized_span_name` attribute
  trace_statements {
    context = "span"
    statements = [
      `set_semconv_span_name("1.37.0", "unsanitized_span_name")`,
    ]
  }

  output {
    traces = [otelcol.connector.spanmetrics.default.input]
  }
}

otelcol.connector.spanmetrics "default" {
  histogram {
    explicit {}
  }

  output {
    metrics = [otelcol.exporter.otlphttp.observability_backend.input]
  }
}

otelcol.exporter.otlphttp "observability_backend" {
  client {
    endpoint = sys.env("OTLP_SERVER_ENDPOINT")
  }
}
```

#### Address high cardinality span names in the instrumentation code

The preferred long-term solution is to ensure span names and attributes comply with [OpenTelemetry Semantic Conventions](https://opentelemetry.io/docs/specs/semconv/) directly in the instrumentation code.

Custom web frameworks are a common source of high cardinality span names.
While default OpenTelemetry instrumentation, for example, Java Servlet, may assign generic span names like `GET /my-web-fwk/*`, your framework has access to more specific routing information.

> By overwriting span attributes in your framework code, you can create compliant, low-cardinality span names that preserve operational granularity.

Reviewer: ```suggestion
Overwrite span attributes in your framework code to create compliant, low-cardinality span names that preserve operational granularity.
```
